pkg/ebpf/tracee.go

```diff
@@ -240,6 +240,30 @@ func New(cfg Config) (*Tracee, error) {
 		}
 	}
 
+	// Clean up signatures not chosen for tracing
+	// Looping from the back to avoid bounds checking
+	for i := len(t.config.EngineConfig.Signatures) - 1; i > 0; i-- {
+		sig := t.config.EngineConfig.Signatures[i]
+		md, _ := sig.GetMetadata()
+		sigEventName := md.EventName
+		sigEventId, ok := events.Definitions.GetID(sigEventName)
+		if !ok {
+			logger.Debugw("removing signature from events to trace", "signature", sigEventName, "reason", "ID does not exist in definitions")
+			t.config.EngineConfig.Signatures = append(t.config.EngineConfig.Signatures[:i], t.config.EngineConfig.Signatures[i+1:]...)
+			continue
+		}
+		_, ok = t.events[sigEventId]
+		if !ok {
+			logger.Debugw("removing signature from events to trace", "signature", sigEventName, "reason", "was not requested for tracing")
+			t.config.EngineConfig.Signatures = append(t.config.EngineConfig.Signatures[:i], t.config.EngineConfig.Signatures[i+1:]...)
+		}
+	}
+
+	// Disable engine pipeline if no signatures are traced post cleanup
+	if len(t.config.EngineConfig.Signatures) == 0 {
```

Comment:
This is probably not correct @josedonizetti.
When we will have Load/Unload event (in v0.16?) then we can possibly run 0 sigs initially then load one later. But if we disable the pipeline stage from the get go, that won't be possible (without adding code to reconfigure and modify the pipeline stages on the go).
So the correct thing is likely keeping the pipeline stage on, but adding a check for sig length before inputting the event to the engine, if no sigs are running we can optimize and skip it into the next one.